controllers/statefulset.go

```diff
@@ -1132,6 +1132,9 @@ func (r *SingleClusterReconciler) updateSTSFromPodSpec(
 	st.Spec.Template.ObjectMeta.Annotations = r.aeroCluster.Spec.PodSpec.AerospikeObjectMeta.Annotations
 
 	st.Spec.Template.Spec.DNSPolicy = r.aeroCluster.Spec.PodSpec.DNSPolicy
+	if r.aeroCluster.Spec.PodSpec.DNSConfig != nil {
```

Comment:
Due to this `if check`, the user will not be able to reset the DNS config to nil.

Reply:
can we make this check over DNSPolicy?

Reply:
Removed if condition to handle both create and update scenario